src/grid/Cell.tsx

```diff
@@ -86,40 +88,52 @@ export default class Cell extends ThemedMixin(FocusMixin(WidgetBase))<CellProper
 	}
 
 	protected render(): DNode {
-		let { editable, rawValue, theme, classes } = this.properties;
+		let { editable, rawValue, theme, classes, width } = this.properties;
 
-		return v('div', { role: 'cell', classes: [this.theme(css.root), fixedCss.rootFixed] }, [
-			this._editing
-				? w(TextInput, {
-						key: 'input',
-						theme,
-						classes,
-						label: `Edit ${rawValue}`,
-						labelHidden: true,
-						extraClasses: { input: this.theme(css.input) } as any,
-						focus: this._focusKey === 'input' ? this.shouldFocus : () => false,
-						value: this._editingValue,
-						onValue: this._onInput,
-						onBlur: this._onBlur,
-						onKeyDown: this._onKeyDown
-				  })
-				: this.renderContent(),
-			editable && !this._editing
-				? w(
-						Button,
-						{
-							key: 'button',
+		return v(
+			'div',
+			{
+				role: 'cell',
+				styles: width
+					? {
+							flex: `0 1 ${width}px`
+					  }
+					: {},
+				classes: [this.theme(css.root), fixedCss.rootFixed]
+			},
+			[
+				this._editing
+					? w(TextInput, {
+							key: 'input',
 							theme,
 							classes,
-							aria: { describedby: this._idBase },
-							focus: this._focusKey === 'button' ? this.shouldFocus : () => false,
-							type: 'button',
-							extraClasses: { root: this.theme(css.edit) } as any,
-							onClick: this._onEdit
-						},
-						[w(Icon, { type: 'editIcon', altText: 'Edit', classes, theme })]
-				  )
-				: null
-		]);
+							label: `Edit ${rawValue}`,
```

Comment:
Maybe this would belong in a separate issue, but these labels should probably use i18n.

Reply:
Yup that should be a separate issue. I've not added this as part of these changes, just the indentation has changed.